ci/Jenkinsfile

```diff
@@ -0,0 +1,93 @@
+@Library('jenkins_shared_lib') _
+
+pipeline {
+  agent any
+  environment {
+    BUILD_MODE = 'pull-request'
+  }
+  options {
+    buildDiscarder(logRotator(numToKeepStr: '10'))
+    ansiColor('xterm')
+  }
+  stages {
+    stage("Style Checks") {
+      steps {
+        sh "env"
+      }
+    }
+    stage("C++ Matrix Build") {
+      environment {
+        PARALLEL_LEVEL = '4'
+      }
+      steps {
+        script {
+          parallel generateStage("cuda_build", {
+            withCredentials([[
+                $class: 'AmazonWebServicesCredentialsBinding',
+                credentialsId: "aws-s3-gpuci",
+                accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+                secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+            ]]) {
+              sh "./ci/build_cpp.sh"
+            }
+          })
+        }
+      }
+    }
+
+    stage("Python Matrix Build") {
+      steps {
+        script {
+          parallel generateStage("python_build", {
+            withCredentials([[
+                $class: 'AmazonWebServicesCredentialsBinding',
+                credentialsId: "aws-s3-gpuci",
+                accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+                secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+            ]]) {
+              sh "./ci/build_py.sh"
+            }
+          })
+        }
+      }
+    }
+
+    stage("PR/Branch Test Stage") {
+      when { not { triggeredBy 'TimerTrigger' } }
+      steps {
+        sh "env"
+        echo "${currentBuild.buildCauses}"
+        script {
+          parallel generateStage("branch_pr_test",{
+            cleanWs (
+              deleteDirs: true,
+              externalDelete: 'sudo rm -rf %s'
+            )
+            checkout scm
+            withCredentials([[
+                $class: 'AmazonWebServicesCredentialsBinding',
+                credentialsId: "aws-s3-gpuci",
+                accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+                secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+            ]]) {
+              sh "./ci/gpu_test.sh"
+            }
+          })
+        }
+      }
+    }
+
+    stage("Nightly Test Stage") {
+      when { triggeredBy 'TimerTrigger' }
+      environment { UPLOAD_PKGS = "true" }
+      steps {
+        echo "${currentBuild.buildCauses}"
+        script {
+          parallel generateStage("nightly_test", {
+            sh "./ci/gpu_test.sh"
+          })
```

Comment:
This step will also need to be wrapped with the `withCredentials` wrapper. I think we should probably just move the `withCredentials` wrapper to the shared library since it will also be used in every stage by every RAPIDS repository. Thoughts?

Reply:
We can prolly decide on that a little later